packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx

```diff
@@ -75,82 +72,44 @@ type TreeNodeRendererProps = Pick<ReturnType<typeof useTree>, "expandNode"> &
-    { node, selected, actions, error, expandNode, getLabel, getSublabel, onNodeClick, onNodeKeyDown, getDecorations, reloadTree, ...treeItemProps },
+  forwardRef<HTMLElement, TreeNodeRendererProps>(function HierarchyNode(
+    { node, selected, error, expandNode, onNodeClick, onNodeKeyDown, reloadTree, getLabel, getSublabel, getActions, getDecorations, ...treeItemProps },
     forwardedRef,
   ) {
     const nodeRef = useRef<HTMLElement>(null);
     const ref = useMergedRefs(forwardedRef, nodeRef);
     const { localizedStrings } = useLocalizationContext();
 
-    const nodeActions = useMemo(() => {
-      const actionButtons: ReactElement[] = [];
-      if (error && error.error.type === "Unknown") {
-        actionButtons.push(
-          <TreeActionButton
-            label={localizedStrings.retry}
-            action={() => reloadTree({ parentNodeId: node.id, state: "reset" })}
-            show={true}
-            icon={refreshSvg}
-          />,
-        );
-      }
-
-      if (!actions || actions.length === 0) {
-        return actionButtons;
-      }
-
-      actionButtons.push(
-        ...actions.map((action, index) => {
-          const actionInfo = action(node);
-          return <TreeActionButton key={index} {...actionInfo} />;
-        }),
-      );
-
-      return actionButtons;
-    }, [actions, error, node, reloadTree, localizedStrings]);
+    const label = useMemo(() => (getLabel ? getLabel(node) : node.label), [getLabel, node]);
+    const description = useMemo(() => (getSublabel ? getSublabel(node) : undefined), [getSublabel, node]);
+    const decorations = useMemo(() => getDecorations?.(node), [getDecorations, node]);
+    const actions = useMemo(
+      () => [
+        ...(error && error.error.type === "Unknown"
+          ? [
+              <Tree.ItemAction
+                key="retry"
+                label={localizedStrings.retry}
+                onClick={() => reloadTree({ parentNodeId: node.id, state: "reset" })}
+                visible={true}
+                icon={refreshSvg}
+              />,
+            ]
+          : []),
+        ...(getActions ? getActions(node) : []),
+      ],
+      [getActions, node, error, localizedStrings, reloadTree],
+    );
 
     const isDisabled = treeItemProps["aria-disabled"] === true;
     return (
       <Tree.Item
         {...treeItemProps}
         ref={ref}
-        aria-level={node.level}
-        aria-posinset={node.posInLevel}
-        aria-setsize={node.levelSize}
-        label={useMemo(() => (getLabel ? getLabel(node) : node.label), [getLabel, node])}
-        description={useMemo(() => (getSublabel ? getSublabel(node) : undefined), [getSublabel, node])}
+        label={label}
+        description={description}
         selected={selected}
-        expanded={node.isExpanded || node.children === true || node.children.length > 0 ? node.isExpanded : undefined}
+        expanded={node.isExpanded || node.children ? node.isExpanded : undefined}
```

Comment:
This seems to add an expander to last nodes with no children.

Reply:
If the node doesn't have children (`node.children` is an empty array), then I guess we don't care about whether the expander is set to expanded or collapsed mode, since it shouldn't be displayed in the first place. However, I'm not sure what controls its display 🤔

Reply:
@MartynasStrazdas Yes you are right. I think this was a leftover from testing different memoization strategies. I will revert it.

Reply:
If we don't want an expander we need to set it to undefined, which in this case does not happen.

Reply:
> `expanded` have to be set to `undefined` in order for expander to be hidden.

wasn't expecting that 🤷‍♂️